### ISSUE TITLE
Support JDK 11

### ISSUE THREAD

**Issue by antfling** [alibaba/java-dns-cache-manipulator#48]
可否支持下jdk11

**Comment by oldratlee** [comment]
Jdk 9及以上的支持 还没做， 😩

@airlove 要不要 来搞一下？ ❤️

**Comment by airlove** [comment]
我今天改了下代码，发现最后由于jdk9以上的modul的封装问题，没有权限修改java.base中的反射，我可以先把代码提交一下
@oldratlee

**Comment by arcuri82** [comment]
Hi!
 I apologize for writing in English, but better than trying to copy&paste here what Google Translate would give...

Anyway, I ended up here because I needed the exact functionality this library provides! Thanks for open-sourcing it! 
But as you might have guessed, I would need support for JDK9+ (just 11 to be specific).

Reflection on java.net gives nasty warnings in JDK 11, but still works. If one does not like the warnings, can use ` --add-opens java.base/java.net=ALL-UNNAMED` as a VM option (ugly, but still better than nothing). Or is there some other issue here that prevents supporting JDK11? 

This issue and PR is 1 year old. So I guess it will not be merged and new release of library will be made? 
That's understandable. But just wanted to double-check with you before spending time re-implementing something equivalent

again, thanks!

**Comment by antfling** [comment]
@arcuri82 
Hi!
Thank you replay this issue.I Test this branch at environment jdk9  cant run.but i test jdk11 can run.this you answer. 
i will review this code and test jdk11 and jdk 13 to run at this week, i would't to support jdk9 or jdk 10. so if you can wait,i will try again.

**Comment by arcuri82** [comment]
excellent! thx! 
I am only interested in LTS versions of the JDK (ie 8 and 11, where the next one will be 17 in a couple of years). I have no interest in 9 and 10... so that's good :)

**Comment by antfling** [comment]
@arcuri82  @oldratlee  https://github.com/alibaba/java-dns-cache-manipulator/pull/51  you can see this PR. the 50 PR because i have change name so you can't see it. i fix some bug and change some test unit case that make new PR. if you want use. you can package yourself.

**Comment by oldratlee** [comment]
@arcuri82 @antfling

Released v1.6.1: JDK 17 support improvement
https://github.com/alibaba/java-dns-cache-manipulator/releases/tag/v1.6.1


add below java options for JDK 17:

```sh
--add-opens java.base/java.net=ALL-UNNAMED
--add-opens java.base/sun.net=ALL-UNNAMED
```

remove `sun.net.util.IPAddressUtil` dependency, simplify `JDK 17` usage, so no need java option `--add-opens java.base/sun.net.util=ALL-UNNAMED`
